contracts/MintShop1155.sol

```diff
@@ -752,12 +753,13 @@ contract MintShop1155 is Sweepable, ReentrancyGuard {
   */
   function getPurchaseCounts(uint256[] calldata _ids,
     address[] calldata _purchasers) external view returns (uint256[][] memory) {
-    uint256[][] memory purchaseCounts;
+    uint256[][] memory purchaseCounts = new uint256[][](_ids.length);
     for (uint256 i = 0; i < _ids.length; i++) {
+      purchaseCounts[i] = new uint256[](_purchasers.length);
       uint256 id = _ids[i];
       for (uint256 j = 0; j < _purchasers.length; j++) {
         address purchaser = _purchasers[j];
-        purchaseCounts[j][i] = pools[id].purchaseCounts[purchaser];
+        purchaseCounts[i][j] = pools[id].purchaseCounts[purchaser];
```

Comment:
Ditto here; I'd prefer we keep the same output order with input parameters as `_purchasers` then `_ids`.